src/Compiler/Compiler.cpp

```diff
@@ -1461,33 +1461,238 @@ bool Compiler::RegisterCommands(TclInterpreter* interp, bool batchMode) {
   interp->registerCmd("open_project", open_project, this, nullptr);
   interp->registerCmd("run_project", run_project, this, nullptr);
 
-  auto gtkwave = [](void* clientData, Tcl_Interp* interp, int argc,
-                    const char* argv[]) -> int {
-    QStringList args{};
-    if (argc > 1) {
-      QString file = QString::fromStdString(argv[1]);
-      if (file.count() > 0 && file[0] == "~") {
-        // QProcess doesn't substitue ~/ paths so we'll manually turn ~ paths
-        // into absolute paths
-        file = QDir::homePath() + file.mid(1);
+  auto wave_cmd = [](void* clientData, Tcl_Interp* interp, int argc,
+                     const char* argv[]) -> int {
+    Compiler* compiler = (Compiler*)clientData;
+    if (compiler) {
+      // Convert args to a single string
+      QStringList args{};
+      for (int i = 1; i < argc; i++) {
+        args.append(argv[i]);
       }
-      args << file;
+      QString cmd = args.join(" ") + "\n";
+
+      // Send cmd to GTKWave
+      compiler->GTKWave_send_cmd(cmd.toStdString());
     }
+    return TCL_OK;
+  };
+  interp->registerCmd("wave_cmd", wave_cmd, this, nullptr);
 
-    auto binPath = GlobalSession->Context()->BinaryPath();
-    auto exePath = binPath / "gtkwave" / "bin" / "gtkwave";
+  auto wave_open = [](void* clientData, Tcl_Interp* interp, int argc,
+                      const char* argv[]) -> int {
+    Compiler* compiler = (Compiler*)clientData;
+    if (compiler) {
+      QString file{};
+      // check if a file was requested
+      if (argc > 1) {
+        file = QString::fromStdString(argv[1]);
+        if (file.count() > 0 && file[0] == "~") {
+          // ~/ doesn't get substitued so we'll manually turn ~ paths into
+          // absolute paths
+          file = QDir::homePath() + file.mid(1);
+        }
+      }
+
+      // if a file was passed, set the loadFile command
+      QString cmd{};
+      if (!file.isEmpty()) {
+        cmd = "gtkwave::loadFile " + file;
+      }
+
+      // Send cmd to GTKWave
+      compiler->GTKWave_send_cmd(cmd.toStdString());
+    }
+    return TCL_OK;
+  };
+  interp->registerCmd("wave_open", wave_open, this, nullptr);
+
+  auto wave_time = [](void* clientData, Tcl_Interp* interp, int argc,
+                      const char* argv[]) -> int {
+    Compiler* compiler = (Compiler*)clientData;
+    if (compiler) {
+      if (argc != 2) {
+        Tcl_AppendResult(
+            interp,
+            qPrintable("Expected Syntax: wave_time <timeString>\ntimeString "
+                       "can specify its time units. Ex: 1ps, 1000ns, 1s"),
+            nullptr);
+        return TCL_ERROR;
+      }
+
+      QString cmd =
```

Comment:
we try do not use QString for the compiler if it is possible. In case tcl ,commands I guess we can use std::string

Reply:
fixed